File: python/Data Structures and Algorithms in Python Book/oop/creditcard.py
class CreditCard:
    
    def __init__(self, customer, bank, accnt, limit):
        self.__customer = customer
        self.__bank = bank
        self.__account= accnt
        self.__limit = limit
        self.__balance = 0

    def get_customer(self):
        return self.__customer
    
    def get_bank(self):
        return self.__bank
    
    def get_account(self):
        return self.__account
    
    def get_limit(self):
        return self.__limit
    
    def get_balance(self):
        return self.__balance
    
    def charge(self, price):
        if price + self.__balance > self.__limit:
            return False
        else:
            self.__balance += price
            return True
    
    def make_payment(self, amount):
        self.__balance -=  amount
    
    def __str__(self):
        return 'Customer: {}, Bank: {}, Account: {}, Limit: {}'.format(
            self.__customer, 
            self.__bank, 
            self.__account, 
            self.__limit)
    
if __name__ == '__main__':
    cc = CreditCard('Gs Su', 'CITI BANK', '1234 5678 9012', 10000)
    print(cc)
    

File: java/primer/CreditCard.java
package primer;

/**
 * A simple model for a consumer credit card.
 * 
 * @author Gaurav
 */
public class CreditCard {
    // Instance variables
    private String customerName;
    private String bankName;
    private String accountId;
    private int limit;
    protected double balance;

    /**
     * Construct a new credit card instance.
     * 
     * @param customerName  the name of the customer
     * @param bankName      the name of the bank
     * @param accountId     the accoundt ID
     * @param limit         the credit limit ( measured in dollars )
     * @param initalBalance the initial balance ( measured in dollars )
     */
    public CreditCard(String customerName, String bankName, String accountId, int limit, double initialBalance) {
        this.customerName = customerName;
        this.bankName = bankName;
        this.accountId = accountId;
        this.limit = limit;
        this.balance = initialBalance;
    }

    /**
     * Construct a new credit card instance with intial balance as zero.
     * 
     * @param customerName the name of the customer
     * @param bankName     the name of the bank
     * @param accountId    the accoundt ID
     * @param limit        the credit limit ( measured in dollars )
     */
    public CreditCard(String customerName, String bankName, String accountId, int limit) {
        this.customerName = customerName;
        this.bankName = bankName;
        this.accountId = accountId;
        this.limit = limit;
        this.balance = 0;
    }

    /**
     * Charges the given price to the card assuming sufficient credit limit.
     * 
     * @param price the amount to be charged
     * @return true if charge was accepted; false otherwise
     */
    public boolean charge(double price) {
        if (price + balance > limit)
            return false;
        balance += price;
        return true;
    }

    /**
     * Process customer payment that reduces balance.
     * 
     * @param amount the amount of payment made
     */
    public void makePayment(double amount) {
        balance -= amount;
    }

    // Accessor methods
    public String getCustomerName() {
        return customerName;
    }

    public String getBankName() {
        return bankName;
    }

    public String getAccountId() {
        return accountId;
    }

    public int getLimit() {
        return limit;
    }

    public double getBalance() {
        return balance;
    }

    @Override
    public String toString() {
        return "CreditCard [accountId=" + accountId + ", balance=" + balance + ", bankName=" + bankName
                + ", customerName=" + customerName + ", limit=" + limit + "]";
    }

    
}
